src/gamestate.lua

```diff
@@ -747,6 +769,131 @@ function read_player_vars(_player_obj)
   read_parry_state(_player_obj.parry_air, _player_obj.parry_air_validity_time_addr, _player_obj.parry_air_cooldown_time_addr)
   read_parry_state(_player_obj.parry_antiair, _player_obj.parry_antiair_validity_time_addr, _player_obj.parry_antiair_cooldown_time_addr)
 
+-- CHARGE STATE
+  -- global game consts
+  _player_obj.charge_1 = _player_obj.charge_1 or { name = "Charge1", max_charge = 42, max_reset = 42, enabled = false }
+  _player_obj.charge_2 = _player_obj.charge_2 or { name = "Charge2", max_charge = 42, max_reset = 42, enabled = false }
+  _player_obj.charge_3 = _player_obj.charge_3 or { name = "Charge3", max_charge = 42, max_reset = 42, enabled = false }
```

Comment:
This could be moved inside the `make_player_object` function in order to avoid unneeded execution